memory_test.go

```diff
@@ -0,0 +1,162 @@
+package gosql
+
+import (
+	"errors"
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+var mb *MemoryBackend
+
+func init() {
+	helpMessagesDisabled = true
+}
+
+func TestSelect(t *testing.T) {
+	mb = NewMemoryBackend()
+	//
+	ast, err := Parse("SELECT * FROM test")
+	if err != nil {
+		panic(err)
+	}
+	assert.NotEqual(t, ast, nil)
+	_, err = mb.Select(ast.Statements[0].SelectStatement)
+	assert.Equal(t, err, errors.New("Table does not exist"))
+	//
+	ast, err = Parse("CREATE TABLE test(x INT, y INT, z INT);")
+	if err != nil {
+		panic(err)
+	}
+	assert.NotEqual(t, ast, nil)
+	err = mb.CreateTable(ast.Statements[0].CreateTableStatement)
+	if err != nil {
+		panic(err)
+	}
+	//
+	ast, err = Parse("INSERT INTO test VALUES(100, 200, 300)")
+	if err != nil {
+		panic(err)
+	}
+	assert.NotEqual(t, ast, nil)
+	err = mb.Insert(ast.Statements[0].InsertStatement)
+	assert.Equal(t, err, nil)
+	//
+	for _, str := range []string{
+		"SELECT * FROM test",
+		"SELECT x FROM test",
+		"SELECT x, y FROM test",
+		"SELECT x, y, z FROM test",
+		"SELECT *, x FROM test",
+		"SELECT *, x, y FROM test",
+		"SELECT *, x, y, z FROM test",
```

Comment:
Good to range over tests but I'd also include the expected values in here too rather than hard-coding them in the for loop